plugin/session_buffer.py

```diff
@@ -496,8 +496,13 @@ def do_document_diagnostic_async(
             return
         if mgr.should_ignore_diagnostics(self._last_known_uri, self.session.config):
             return
+        change_count = view.change_count()
         if version is None:
-            version = view.change_count()
+            version = change_count
+        elif version < change_count:
+            return
+        if version == self._diagnostics_version:
+            return
```

Comment:
I've identified cases (by furiously editing the document and undoing changes) where this code would run with `version` that is already behind `view.change_count()`. Thus the added `version < change_count` check. This is because there can be some delay between document change and this triggering.